### ISSUE TITLE
Payment hangs when trying to make a direct payment

### ISSUE THREAD

**Issue by nephix** [raiden-network/light-client#376]
## Steps to Reproduce

1. Open channel with a hub
2. Deposit some tokens
3. Attempt to send payment to hub
4. Confirm MetaMask transactions

## Expected Result
- Payment goes through

## Actual Result
- Dapp indefinitely shows payment screen w/ spinner

## Additional Infos (Optional) 
In this instance I attempted a direct payment from `0x919937afdBD25a5F0f92317b51C6205978f759F0` to `0x456e2459b6E01FF30375D59f24d359655bEB6754`.

Console outputs no errors, but Matrix Client says:
> sendEvent of type m.room.message in !UZFYCOSVZcnNTGAgNl:transport01.raiden.network with txnId m1570532445109.8
> Event sent to !UZFYCOSVZcnNTGAgNl:transport01.raiden.network with event id $FKKLzghoyLXUdTShdNmacytJHnRnaegbz18-bnsfSt0
> setting pendingEvent status to sent in !UZFYCOSVZcnNTGAgNl:transport01.raiden.network
> Already have event $L9P-RSQKXUWE1gzvXORceuvxGHv3j5_Fyf7yxH4YTg0 in limited sync - not resetting

These occur repeatedly as long as I keep the dapp running on the payment screen.

In the raiden client, I'm getting repeatedly:

> 2019-10-08 10:58:47.008473 [warning  ] Message already in queue - ignoring [raiden.network.transport.matrix.transport] current_user=@0x456e2459b6e01ff30375d59f24d359655beb6754:transport01.raiden.network message=<Processed [msghash=0x350d476fa457bc2]> node=0x456e2459b6E01FF30375D59f24d359655bEB6754 queue=QueueIdentifier(recipient=b'\xb8Lk\xfb\xa1\x10\x85C\xeaG\xe8\xb9\xa6i\xcb9\xff\xc7\x0fj', canonical_identifier=CanonicalIdentifier(chain_identifier=0, token_network_address=b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', channel_identifier=0)) receiver=0xB84C6bFBA1108543Ea47e8B9a669CB39ffc70f6A transport_uuid=93d83681-bf73-4c77-a0b5-8547ef42c4bc

Dapp Local Storage:
<details><summary>CLICK ME</summary>
<p>

```
{
  "address": "0x919937afdBD25a5F0f92317b51C6205978f759F0",
  "chainId": 5,
  "registry": "0x754b1Fa6bcb43F09557Eb5A212513F40A263DAF8",
  "blockNumber": 1434283,
  "channels": {
    "0xE36995596D456aaEcD226a303250Bf99cf16b2A0": {
      "0x691A2E94E4A2b6178db98b92Ea13E9ddE1695CC8": {
        "state": "open",
        "own": {
          "deposit": 90000000000000
        },
        "partner": {
          "deposit": 400000000000000
        },
        "id": 33,
        "settleTimeout": 500,
        "openBlock": 1433278,
        "isFirstParticipant": true
      },
      "0x456e2459b6E01FF30375D59f24d359655bEB6754": {
        "state": "open",
        "own": {
          "deposit": 2400000000000000,
          "locks": [
            {
              "amount": 90000000000,
              "expiration": 1434337,
              "secrethash": "0x08b3e35ed2bbf51ec0a92659a017935659b7cd3d64fc09a4b96f9518da0e82c1"
            }
          ],
          "balanceProof": {
            "chainId": 5,
            "tokenNetworkAddress": "0xE36995596D456aaEcD226a303250Bf99cf16b2A0",
            "channelId": 35,
            "nonce": 1,
            "transferredAmount": 0,
            "lockedAmount": 90000000000,
            "locksroot": "0xdc3817fe00b0a43264946fe11355ad56a9f7977abe9ebee313a3570ef88676c7",
            "messageHash": "0xc51e778d98ced5638237ea9014cc3b394c0b99ceadd7511bc8894eec1569b85b",
            "signature": "0x25ad50f6a63355048a31e00819d5cde767ca9b63d402fdb20f41aac0346048d758c69d1000a7c834feb811caf74eaba5abdeda5738ffdfc28384c765cc1c853c1c",
            "sender": "0x919937afdBD25a5F0f92317b51C6205978f759F0"
          }
        },
        "partner": {
          "deposit": 0
        },
        "id": 35,
        "settleTimeout": 500,
        "openBlock": 1433846,
        "isFirstParticipant": true
      }
    }
  },
  "tokens": {
    "0x3ed0DaEDC3217615bde34FEDd023bC81ae49251B": "0xE36995596D456aaEcD226a303250Bf99cf16b2A0",
    "0x62083c80353Df771426D209eF578619EE68D5C7A": "0x9526EC0E58ac5Bb1e24d655E964f0eAA46054d16",
    "0x95B2d84De40a0121061b105E6B54016a49621B44": "0x70Ca98F397e48FA605886eCb276eCd04F9136165"
  },
  "transport": {
    "matrix": {
      "server": "https://transport03.raiden.network",
      "setup": {
        "userId": "@0x919937afdbd25a5f0f92317b51c6205978f759f0:transport03.raiden.network",
        "accessToken": "MDAyOGxvY2F0aW9uIHRyYW5zcG9ydDAzLnJhaWRlbi5uZXR3b3JrCjAwMTNpZGVudGlmaWVyIGtleQowMDEwY2lkIGdlbiA9IDEKMDA1OWNpZCB1c2VyX2lkID0gQDB4OTE5OTM3YWZkYmQyNWE1ZjBmOTIzMTdiNTFjNjIwNTk3OGY3NTlmMDp0cmFuc3BvcnQwMy5yYWlkZW4ubmV0d29yawowMDE2Y2lkIHR5cGUgPSBhY2Nlc3MKMDAyMWNpZCBub25jZSA9IFlQZG9Sci1aLVBDcmxzeWMKMDAyZnNpZ25hdHVyZSB2n_AcwKqrHvVl3nZI8dCMGPnaSJcyNdPxPhaZrR8fGwo",
        "deviceId": "ZJJIXSMKLX",
        "displayName": "0x4f1d185fa1cf495aab3567a0495ec1a5d79515563e4d589dc5e4f4e6176387dd5410c99d9e48063ed5dfa257eff702cd8bc42c4ff70b59ba0d191879e7af0bee1b"
      },
      "rooms": {
        "0x456e2459b6E01FF30375D59f24d359655bEB6754": [
          "!UZFYCOSVZcnNTGAgNl:transport01.raiden.network"
        ],
        "0x691A2E94E4A2b6178db98b92Ea13E9ddE1695CC8": [
          "!VBocRNLIsxmDFzwODx:transport01.raiden.network"
        ]
      }
    }
  },
  "secrets": {
    "0x08b3e35ed2bbf51ec0a92659a017935659b7cd3d64fc09a4b96f9518da0e82c1": {
      "secret": "0xeac3f8c3a9ad2fb8d9987c956ffd99b73fa1e554a2b096f0a02e08c845dec104"
    }
  },
  "sent": {
    "0x08b3e35ed2bbf51ec0a92659a017935659b7cd3d64fc09a4b96f9518da0e82c1": {
      "transfer": [
        1570532200729,
        {
          "type": "LockedTransfer",
          "message_identifier": 1570532197898,
          "chain_id": 5,
          "token_network_address": "0xE36995596D456aaEcD226a303250Bf99cf16b2A0",
          "channel_identifier": 35,
          "nonce": 1,
          "transferred_amount": 0,
          "locked_amount": 90000000000,
          "locksroot": "0xdc3817fe00b0a43264946fe11355ad56a9f7977abe9ebee313a3570ef88676c7",
          "payment_identifier": 223752234096379353,
          "token": "0x3ed0DaEDC3217615bde34FEDd023bC81ae49251B",
          "recipient": "0x456e2459b6E01FF30375D59f24d359655bEB6754",
          "lock": {
            "amount": 90000000000,
            "expiration": 1434337,
            "secrethash": "0x08b3e35ed2bbf51ec0a92659a017935659b7cd3d64fc09a4b96f9518da0e82c1"
          },
          "target": "0x456e2459b6E01FF30375D59f24d359655bEB6754",
          "initiator": "0x919937afdBD25a5F0f92317b51C6205978f759F0",
          "fee": 0,
          "metadata": {
            "routes": [
              {
                "route": [
                  "0x456e2459b6E01FF30375D59f24d359655bEB6754"
                ]
              }
            ]
          },
          "signature": "0x25ad50f6a63355048a31e00819d5cde767ca9b63d402fdb20f41aac0346048d758c69d1000a7c834feb811caf74eaba5abdeda5738ffdfc28384c765cc1c853c1c"
        }
      ]
    }
  }
}
```

</p>
</details>

**Comment by christianbrb** [comment]
I get this now as well

Sending from to 0x691A2E94E4A2b6178db98b92Ea13E9ddE1695CC8

After 
```
action transferSecret @ 13:18:20.926
action transferSigned @ 13:18:20.931

```
it hangs

```
action messageSend @ 13:18:25.196
client.js:2057 sendEvent of type m.room.message in !HjkFjChsWWfHIzZuAE:transport03.raiden.network with txnId m1570533505201.78
client.js:2248 Event sent to !HjkFjChsWWfHIzZuAE:transport03.raiden.network with event id $wDx3CLfNd0eY1PlJBRUakUn4_8Lo4VTC1xzLZTtg7Ls
setting pendingEvent status to sent in !HjkFjChsWWfHIzZuAE:transport03.raiden.network
```

```
action messageSent @ 13:18:25.345
sync.js:89 Already have event $wDx3CLfNd0eY1PlJBRUakUn4_8Lo4VTC1xzLZTtg7Ls in limited sync - not resetting
```

**Comment by christianbrb** [comment]
Direct payment to `0x456e2459b6E01FF30375D59f24d359655bEB6754 ` worked

**Comment by andrevmatos** [comment]
It doesn't _hang indefinitely_, just while the payment is pending (which can take by default 2*reveal timeout = 100 blocks), and is expected. From your logs, seems to be something wrong on raiden-py side, the message was sent and accepted (Processed on their queue), but the responses didn't come through. It'll eventually expire and be cleared on our side.

**Comment by nephix** [comment]
> It doesn't hang indefinitely

The UI definitely does.

**Comment by kelsos** [comment]
> > It doesn't hang indefinitely
> 
> The UI definitely does, and I doubt that this is expected.

The way it is currently implemented the UI will block until the payment fails, if it fails on lock expiry then this blocking will be in ~2 hours. 

@nephix Sure this is not optimal and we already discussed with @andrevmatos when the `transfer` method was updated about having some kind of list of pending transfers and notifications instead of blocking. 

I am not sure how the user experience will be for such an approach though, at least while we still have to interactively sign each message.
